xcode/Subconscious/SubconsciousTests/Tests_Sphere.swift

```diff
@@ -0,0 +1,270 @@
+            guard let memo = sphere.read(slashlink: "/foo") else {
+                XCTFail("Could not read memo")
+                return
+            }
+            XCTAssertEqual(memo.contentType, "text/subtext")
+        }
+    }
+    
+    func testHeadersRoundtrip() throws {
+        let noosphere = noosphere!
+        let sphereReceipt = try noosphere.createSphere(ownerKeyName: "bob")
+        
+        let sphere = try Sphere(
+            noosphere: noosphere,
+            identity: sphereReceipt.identity
+        )
+        
+        let then = Date.distantPast.ISO8601Format()
+        try sphere.write(
+            slug: "foo",
+            contentType: "text/subtext",
+            additionalHeaders: [
+                Header(name: "Title", value: "Foo"),
+                Header(name: "Created", value: then),
+                Header(name: "Modified", value: then),
+                Header(name: "File-Extension", value: "subtext")
+            ],
+            body: "Test".toData(encoding: .utf8)!
+        )
+        _ = try sphere.save()
+        guard let memo = sphere.read(slashlink: "/foo") else {
+            XCTFail("Could not read memo")
+            return
+        }
+        XCTAssertEqual(memo.contentType, "text/subtext")
+        XCTAssertEqual(memo.body.toString(), "Test")
+        
+        let title = memo.additionalHeaders.get(first: "Title")
+        XCTAssertEqual(title, "Foo")
+        
+        let ext = memo.additionalHeaders.get(first: "File-Extension")
+        XCTAssertEqual(ext, "subtext")
+        
+        let created = memo.additionalHeaders.get(first: "Created")
+        XCTAssertEqual(created, then)
+        
+        let modified = memo.additionalHeaders.get(first: "Created")
+        XCTAssertEqual(modified, then)
+    }
+    
+    func testList() throws {
+        let noosphere = noosphere!
+        let sphereReceipt = try noosphere.createSphere(ownerKeyName: "bob")
+        
+        let sphere = try Sphere(
+            noosphere: noosphere,
+            identity: sphereReceipt.identity
+        )
+        
+        let body = "Test".toData(encoding: .utf8)!
+        
+        try sphere.write(
+            slug: "foo",
+            contentType: "text/subtext",
+            body: body
+        )
+        try sphere.write(
+            slug: "bar",
+            contentType: "text/subtext",
+            body: body
+        )
+        try sphere.write(
+            slug: "baz",
+            contentType: "text/subtext",
+            body: body
+        )
+        
+        let slugsA = try sphere.list()
+        XCTAssertEqual(
+            slugsA,
+            [],
+            "Lists all slugs in latest version of sphere"
+        )
+        
+        _ = try sphere.save()
+        
+        let slugsB = try sphere.list()
+        XCTAssertTrue(slugsB.contains("foo"))
+        XCTAssertTrue(slugsB.contains("bar"))
+        XCTAssertTrue(slugsB.contains("baz"))
+    }
+    
+    func testChanges() throws {
```

Comment:
This test is currently failing. Changes always seem to be for-all-time, even when passing a `since`.